mesonbuild/options.py

```diff
@@ -473,8 +474,65 @@ def add_to_argparse(self, name: str, parser: argparse.ArgumentParser, help_suffi
     OptionKey('purelibdir', module='python'): {},
 }
 
-
 class OptionStore:
     def __init__(self):
-        # This class will hold all options for a given build directory
-        self.dummy = None
+        self.d: T.Dict['OptionKey', 'UserOption[T.Any]'] = {}
+
+    def __len__(self):
+        return len(self.d)
+
+    def ensure_key(self, key: T.Union[OptionKey, str]) -> OptionKey:
+        if isinstance(key, str):
+            return OptionKey(key)
+        return key
+
+    def get_value_object(self, key: T.Union[OptionKey, str]) -> 'UserOption[T.Any]':
+        return self.d[self.ensure_key(key)]
+
+    def get_value(self, key: T.Union[OptionKey, str]) -> 'T.Any':
+        return self.get_value_object(key).value
+
+    def add_system_option(self, key: T.Union[OptionKey, str], valobj: 'UserOption[T.Any]'):
+        key = self.ensure_key(key)
+        self.d[key] = valobj
+
+    def add_project_option(self, key: T.Union[OptionKey, str], valobj: 'UserOption[T.Any]'):
+        key = self.ensure_key(key)
+        self.d[key] = valobj
+
+    def set_value(self, key: T.Union[OptionKey, str], new_value: 'T.Any') -> bool:
+        key = self.ensure_key(key)
+        return self.d[key].set_value(new_value)
+
+    # FIXME, this should be removed.
```

Comment:
Is this FIXME still relevant?

Reply:
Yes. There should be separate methods for A) creating a new option with the given option and B) changing the type of an existing option. Due to reasons I can't remember I needed this but which will go away after some more refactoring.

In the end it might just need renaming, not removing so updating.